common/src/main/java/com/wynntils/wrappedscreens/trademarket/TradeMarketSearchResultHolder.java

```diff
@@ -0,0 +1,370 @@
+
+        // We only use set slot events to get the items,
+        // but we don't want the items to be set on our custom screen
+        event.setCanceled(true);
+
+        // Reset the empty item count,
+        // set content packets mean we are on a new page
+        pageItemCount = 0;
+    }
+
+    @SubscribeEvent
+    public void onContainerSetSlot(ContainerSetSlotEvent.Pre event) {
+        if (pageLoadingMode == PageLoadingMode.NONE || requestedPage == -1) return;
+
+        WrappedScreenInfo wrappedScreenInfo = wrappedScreen.getWrappedScreenInfo();
+        if (event.getContainerId() != wrappedScreenInfo.containerId()) return;
+
+        int slot = event.getSlot();
+
+        // We don't want the items to be set on our custom screen
+        event.setCanceled(true);
+
+        ItemStack itemStack = event.getItemStack();
+
+        handleSetItem(slot, itemStack);
+        updateDisplayItems(wrappedScreen.getSearchQuery());
+    }
+
+    @Override
+    protected Pattern getReplacedScreenTitlePattern() {
+        return TITLE_PATTERN;
+    }
+
+    @Override
+    protected TradeMarketSearchResultScreen createWrappedScreen(WrappedScreenInfo wrappedScreenInfo) {
+        return new TradeMarketSearchResultScreen(wrappedScreenInfo, this);
+    }
+
+    @Override
+    protected void setWrappedScreen(TradeMarketSearchResultScreen wrappedScreen) {
+        this.wrappedScreen = wrappedScreen;
+
+        // Preload the first batch of pages
+        loadItemsUntilPage(PAGE_BATCH_SIZE, false);
+    }
+
+    @Override
+    protected void reset() {
+        requestedPage = -1;
+        requestedItem = null;
+        pageLoadingMode = PageLoadingMode.NONE;
+        queuedAction = null;
+        allPagesLoaded = false;
+        currentPage = 0;
+        itemMap = new HashMap<>();
+        pageItemCount = 0;
+        filteredItems = new ArrayList<>();
+
+        this.wrappedScreen = null;
+    }
+
+    public void clickOnItem(ItemStack clickedItem) {
+        // When writing this, items only update on page change,
+        // so if we don't switch pages, we can just click on the item
+        Int2ObjectMap.FastEntrySet<ItemStack> currentPageEntries =
+                itemMap.getOrDefault(currentPage, new Int2ObjectOpenHashMap<>()).int2ObjectEntrySet();
+        for (Int2ObjectMap.Entry<ItemStack> entry : currentPageEntries) {
+            if (ItemUtils.isItemEqual(entry.getValue(), clickedItem)) {
+                // Item found on the current page, click on it
+                WrappedScreenInfo wrappedScreenInfo = wrappedScreen.getWrappedScreenInfo();
+                ContainerUtils.clickOnSlot(
+                        entry.getIntKey(),
+                        wrappedScreenInfo.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                        wrappedScreenInfo.containerMenu().getItems());
+
+                return;
+            }
+        }
+
+        // Assume the item is before the current page
```

Comment:
I don't understand this comment?

Reply:
"We didn't find the item on the current page, **assume it is on the page before the currently loaded one**"

I will try to fix this comment in code too.